Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -2,285 +2,69 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 import struct Foundation.Data
-import class Foundation.InputStream
-import class Foundation.FileManager
-import class Foundation.FileHandle
-import class Foundation.OutputStream
-import struct Foundation.URL
 import AwsCIo
-import AwsCCommon
 import AwsCCal
-#if os(Linux)
-     import Glibc
- #else
-     import Darwin
- #endif
 
-// swiftlint:disable identifier_name superfluous_disable_command
-// Todo: this is being used. Refactor
-public class ByteBuffer: Codable {
+public class ByteBuffer {
```

Comment:
This class is used in the SDK. Orginallay it was an [uint8] buffer with a current index and conforms to IStreamable protocol. I have done the following the changes. Ideally this should have been an extension over the data class to conform to IStreamable protocol. However, because this is being used in SDK, I haven’t completely nuked it.

1. Use Data instead of [uint8] as this is more natural to Swift for ByteBuffer in memory. There is an ByteBuffer implementation but that belongs to SwiftNIO package so we can't use it. Another advantage is the Swift copy on write optimization by using Data directly. 
2. Remove all the endiness & putInt32 etc functions. As per my understanding, they are not used in the SDK and add needless complexity to this class.

Reply:
I have discussed it with @epau and SDK is using byte buffer only at [here](https://github.com/awslabs/aws-sdk-swift/blob/f24c44bd8788aff87a75063639f365e38794bf9c/AWSClientRuntime/Sources/Middlewares/Sha256TreeHashMiddleware.swift#L60)